### PR TITLE
fix column cleaning and general cleanup

### PR DESCRIPTION
**Pull request by shouples**
- enforce datatypes after sampling
- remove unnecessary column cleaning when comparing dataframes
- remove lingering callout code
- more adjustments to debug logging
- add tests for formatting dataframes with missing values (`None`, `np.nan`, `pd.NA`)

known issue:
- conversion to original dtypes after resampling/filtering may not be successful on some types (`pd.Period`/`pd.Interval`) since pandas can't parse a stringified list, so we might need to keep track of some intermediate normalized set of dtypes instead of the "original" dtypes

### CODE REVIEW

File: dx/utils/tracking.py [hunk 1]
```diff
@@ -120,15 +96,28 @@ def get_df_variable_name(
     """
     Returns the variable name of the DataFrame object.
     """
+    logger.debug("looking for matching variables for dataframe")
```
Comment (willingc) [inline dx/utils/tracking.py:99]: Curious. What is the use case for this utility?

File: dx/utils/formatting.py [hunk 2]
```diff
@@ -138,39 +138,11 @@ def clean_column_values_for_display(s: pd.Series) -> pd.Series:
...
-    s = datatypes.handle_complex_number_series(s)
-    s = datatypes.handle_ip_address_series(s)
-
-    s = date_time.handle_time_period_series(s)
-
     s = geometry.handle_geometry_series(s)
 
     s = datatypes.handle_dict_series(s)
     s = datatypes.handle_sequence_series(s)
+    s = datatypes.handle_unk_type_series(s)
```
Comment (willingc) [inline dx/utils/formatting.py:145]: Why is the type unknown for these?

**Review by willingc — APPROVED** [review]
Thanks for the updated tests. I had a couple of questions but those shouldn't block merge.

File: dx/utils/tracking.py (hunk 1 above)
Comment (shouples) [reply dx/utils/tracking.py:99]: This was previously for the blueprintjs callout to the user after a resample request that returned instructions to pass the filtered subset to a new variable, like:
```
new_df = my_awesome_df.query( < pandas query string >, engine="python")
```
But it's going to be passed in the metadata to the frontend to let DEX handle displaying the query syntax dynamically when the filters change, instead of just on (backend) resample requests. I'd like to have this handled in the dataframe caching branch for easier access.

File: dx/utils/formatting.py (hunk 2 above)
Comment (shouples) [reply dx/utils/formatting.py:145]: Mostly as a catch-all for if we run into objects that aren't JSON serializable that we haven't handled in an explicit way earlier, to avoid breaking the pandas `build_json_schema()` call. Definitely open to suggestions here!
